AppInspector.RulesEngine/AbstractRuleSet.cs

```diff
@@ -77,7 +78,14 @@ public IEnumerable<ConvertedOatRule> GetUniversalRules()
         var clauses = new List<Clause>();
         var clauseNumber = 0;
         var expression = new StringBuilder("(");
+
         foreach (var pattern in rule.Patterns)
+        {
+            if (EnableNonBacktrackingRegex && !pattern.Modifiers.Contains("b") && !pattern.Modifiers.Contains("nb"))
```

Comment:
This is an interesting edge case we probably want to some how catch in the validator. B and nb are exclusive options.

Reply:
I also wonder, if `nb` is being added here, do we also need the parameter to `RegexModifierToRegexOptions` or is this nb enough to flow through?

Reply:
Another comment on this line. `b` and `nb` should be checked case insensitively to match implementation in `RegexModifierToRegexOptions`.

Perhaps

`pattern.Modifiers.Any(modifier => modifier.Equals("b", StringComparison.CurrentCultureIgnoreCase)`

Reply:
I rewrote this piece. I'm using string comparison InvariantCultureIgnoreCase similar to all cases in this file. Just keeping the same style.